### PR TITLE
Add tour activity

### PR DESCRIPTION
**Pull request by stblr**

### CODE REVIEW

File: app/src/main/res/layout/tour_entry.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            android:id="@+id/tourDistance"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="5dp"
+            app:layout_constraintStart_toEndOf="@+id/tourDistanceIcon"
+            app:layout_constraintEnd_toStartOf="@id/tourLengthIcon"
+            app:layout_constraintTop_toBottomOf="@id/tourName"
+            app:layout_constraintBottom_toBottomOf="parent" />
+
+        <ImageView
```
Comment (amattei777) [inline app/src/main/res/layout/tour_entry.xml:79]: the contentDescription warnings on images are there

**Review by amattei777 — CHANGES_REQUESTED** [review]
Add call to GuessLocationTourMode that will be merged soon and it will be ok.

File: app/src/androidTest/java/com/github/wnder/TourActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        onView(withText("2")).check(matches(isDisplayed()));
+        onView(withText("3")).check(matches(isDisplayed()));
+
+        // Distance to first picture
+        onView(withText("24 km")).check(matches(isDisplayed()));
+        onView(withText("42 km")).check(matches(isDisplayed()));
+
+        // Length of tour
+        onView(withText("25 km")).check(matches(isDisplayed()));
+        onView(withText("52 km")).check(matches(isDisplayed()));*/
```
Comment (leonardvaney) [inline app/src/androidTest/java/com/github/wnder/TourActivityTest.java:110]: I guess it is no working? If so, you can remove it, but it doesn't seem to be wrong...

File: app/src/main/java/com/github/wnder/tour/TourAdapter.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            holder.getDistanceTextView().setText(dText);
+        });
+
+        tourDb.getTourLength(tourId).thenAccept(tourLength -> {
+            String lText = context.getString(R.string.distance_meter, tourLength.intValue());
+            if (tourLength > 10000) {
+                lText = context.getString(R.string.distance_kilometer, tourLength.intValue() / 1000);
+            }
+            holder.getLengthTextView().setText(lText);
+        });
```
Comment (leonardvaney) [inline app/src/main/java/com/github/wnder/tour/TourAdapter.java:85]: There is a code climate issue that you can easily remove here.

File: app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java [hunk 4]
```diff
@@ -274,4 +277,36 @@ protected String selectImageBasedOnKarma(Map<String, Long> idsAndKarma){
...
+
+                Location tourLocation = new Location("");
+                tourLocation.setLatitude(latitude);
+                tourLocation.setLongitude(longitude);
+
+                double distance = tourLocation.distanceTo(userLocation);
+                distances.add(distance);
+            }
+
+            // Keep the 10 closest tours
```
Comment (leonardvaney) [inline app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java:304]: The comment is not correct anymore.

**Review by leonardvaney — CHANGES_REQUESTED** [review]
The PR is not linked with the GuessLocationActivity at the moment where I'm doing this review, so I'm waiting this to approve.

**Review by amattei777 — APPROVED** [review]

**Review by leonardvaney — APPROVED** [review]

**Pull request merged**